src/Material.cpp

```diff
@@ -1518,3 +1619,230 @@ Material* Material::clone(){
+ *          scattering cross-section within SIGMA_T_THRESH then this method
+ *          exits OpenMOC.
+ */
+void IsoMaterial::checkSigmaT() {
+  if (_num_groups == 0)
+    log_printf(ERROR, "Unable to verify Material %d's total cross-section "
+              "since the number of energy groups has not been set", _id);
+              
+  if (_num_isotopes == 0)
+    log_printf(ERROR, "Unable to verify Material %d's total cross-section "
+              "since no isotopes have been added.", _id);
+
+  FP_PRECISION calc_sigma_t;
+
+  /* Loop over all energy groups */
+  for (int i=0; i < _num_groups; i++) {
+
+    /* Initialize the calculated total xs to the absorption xs */
+    calc_sigma_t = _sigma_a[i];
+
+    /* Increment calculated total xs by scatter xs for each energy group */
+    for (int j=0; j < _num_groups; j++)
+      calc_sigma_t += getSigmaSByGroup(i+1,j+1);
+
+    /* Check if the calculated and total match up to certain threshold */
+    if (fabs(calc_sigma_t - _sigma_t[i]) > SIGMA_T_THRESH) {
+      log_printf(ERROR, "Material id = %d has a different total cross-section "
+                 "than the sum of its scattering and absorption cross-sections "
+                 "for group %d: sigma_t = %f, calc_sigma_t = %f",
+                 _id, i+1, _sigma_t[i], calc_sigma_t);
+    }
+  }
+  
+  return;
+}
+  
+  
+/**
+ * @brief Get the group to group scatter cross section for given groups
+ * @details Sums the contribution from each isotope
+ * @param origin the column index of the scattering matrix
+ * @param destination the row index of the scatteirng matrix
+ * @return the group to group scatter cross section
+ */     
+FP_PRECISION IsoMaterial::getScatterSource(int group, FP_PRECISION* flux) {
+  FP_PRECISION Q = 0;
+  
+  /* sum scatter source for each isotope */
+  for (int i=0; i < _num_isotopes; i++)
+      Q += _isotopes[i]->getScatterSource(group,flux)*_num_dens[i];
+  
+  return Q;
+}
+
+
+/**
+ * @brief Get the group to group scatter cross section for given groups
+ * @details Sums the contribution from each isotope
+ * @param origin the column index of the scattering matrix
+ * @param destination the row index of the scatteirng matrix
+ * @return the group to group scatter cross section
+ */  
+FP_PRECISION IsoMaterial::getSigmaSByGroup(int origin, int destination) {
+  FP_PRECISION xs = 0;
+
+  /* sum scatter xs for each isotope */
+  for (int i=0; i < _num_isotopes; i++)
+      xs += _isotopes[i]->getSigmaSByGroup(origin,destination)*_num_dens[i];
+  
+  return xs;
+}
+
+
+/**
+ * @brief Get the number of isotopes in the Material.
+ * @return the number of isotopes
+ */    
+int IsoMaterial::getNumIsotopes() {
+  return _num_isotopes;
+}
+
+
+/**
+ * @brief Create a duplicate of the IsoMaterial.
+ * @return a pointer to the clone
+ */    
+IsoMaterial* IsoMaterial::clone() {
+  IsoMaterial* clone = new IsoMaterial(getId(),getName());
+  
+  /* Add each isotope in original Material to cloned Material */
```

Comment:
All of the items it inherits are populated by adding the 'Isotopes'.  The only questionable part is the aligned data, which we can talk about when we fix the `GPUSolver` and test the SIMD solver.